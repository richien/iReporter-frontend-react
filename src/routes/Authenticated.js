import React from 'react';
import { Redirect } from 'react-router-dom';

const Authenticated = Component => props =>
  sessionStorage.getItem('isLoggedIn') ? <Component {...props} /> : <Redirect to="/login" />;

export default Authenticated;
