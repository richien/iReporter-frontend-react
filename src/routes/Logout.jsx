import { Component } from 'react';

export class Logout extends Component {
  render() {
    sessionStorage.clear();
    return window.location.replace('/');
  }
}

export default Logout;
