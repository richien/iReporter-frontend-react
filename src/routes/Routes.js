/* eslint-disable import/no-named-as-default */
import React from 'react';
import { Route, BrowserRouter, Switch } from 'react-router-dom';
import LoginPage from '../components/containers/loginPage/LoginPage';

export const Routes = () => (
  <BrowserRouter>
    <div>
      <Switch>
        {/* <Route exact path="/" component={LoginPage} /> */}
        <Route path="/login" component={LoginPage} />
      </Switch>
    </div>
  </BrowserRouter>
);

export default Routes;
