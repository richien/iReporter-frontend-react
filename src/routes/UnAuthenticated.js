import React from 'react';
import { Redirect } from 'react-router-dom';

const UnAuthenticated = Component => props =>
  !sessionStorage.getItem('isLoggedIn') ? <Component {...props} /> : <Redirect to="/incidents" />;

export default UnAuthenticated;
