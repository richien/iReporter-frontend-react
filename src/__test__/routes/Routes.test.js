import React from 'react';
import { shallow } from 'enzyme';
import { Routes } from '../../routes/Routes';

describe('Routes', () => {
  it('should render Routes component', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper).toMatchSnapshot();
  });
});
