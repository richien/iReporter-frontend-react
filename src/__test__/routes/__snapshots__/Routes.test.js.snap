// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render Routes component 1`] = `ShallowWrapper {}`;
